babel.config.js

```diff
@@ -25,52 +25,218 @@
 
 'use strict';
 
-const minimist = require('minimist');
+const argv = require('minimist')(process.argv.slice(2));
+const experimentsConfig = require('./build-system/global-configs/experiments-config.json');
+const experimentsConstantBackup = require('./build-system/global-configs/experiments-const.json');
 const {isTravisBuild} = require('./build-system/common/travis');
-const argv = minimist(process.argv.slice(2));
-
-const isClosureCompiler =
-  argv._.includes('dist') || argv._.includes('check-types');
-const {esm} = argv;
-const noModuleTarget = {
-  'browsers': isTravisBuild()
-    ? ['Last 2 versions', 'safari >= 9']
-    : ['Last 2 versions'],
+
+const jsxOpts = {
+  pragma: 'Preact.createElement',
+  pragmaFrag: 'Preact.Fragment',
+  useSpread: true,
 };
 
-const plugins = [
-  './build-system/babel-plugins/babel-plugin-transform-fix-leading-comments',
+const localPlugin = name =>
+  require.resolve(`./build-system/babel-plugins/babel-plugin-${name}`);
+
+const defaultPlugins = isEsmBuild => [
+  localPlugin('transform-fix-leading-comments'),
   '@babel/plugin-transform-react-constant-elements',
-  [
-    '@babel/plugin-transform-react-jsx',
-    {
-      pragma: 'Preact.createElement',
-      pragmaFrag: 'Preact.Fragment',
-      useSpread: true,
-    },
-  ],
+  ['@babel/plugin-transform-react-jsx', jsxOpts],
+  localPlugin('transform-inline-configure-component'),
```

Comment:
`build.conf` is only used in `dist` and `check-types`. This plugin and below should not be used for `test`.

Reply:
Good catch. Some context on testing below: 

This is how Karma is set up to run `babelify` transforms on test code (using `browserify`):

https://github.com/ampproject/amphtml/blob/3488672872a98df3df29a150bffc7bfca7994586/build-system/tasks/karma.conf.js#L75-L83

It should be possible to [enable debugging](https://github.com/browserify/browserify#browserifyfiles--opts) for `browserify` and run `gulp unit --files <some file>` to see the exact config being picked up by `babelify`.

Reply:
Would you mind clarifying what about this is borked, given that it is unchanged from when it lived in `build.conf.js`? This `defaultPlugins` method is not used for testing AFAICT. `karma.conf.js` gets `BABELIFY_PLUGINS` from `helpers.js`, which looks like this:

```javascript
const BABELIFY_PLUGINS = {
  plugins: [
    conf.getReplacePlugin(argv.esm || false),
    conf.getJsonConfigurationPlugin(),
  ],
};
```

Unless I'm missing something, this PR should be a no-op. As I mention elsewhere I'm happy to clean up/refactor this and anything else (in separate PRs), but I can't tell how this change would have downstream impacts.

Also @rsimha I'm not sure what you were trying to link to there. AFAICT there's no flag that does anything like increase verbosity or display what config is picked up by `babelify`; here, `debug` is already set and all that does is add a source map. I could just log the `BABELIFY_CONFIG`, but It can tell how that's constructed anyway. Was there something specific you had in mind or was your intent to more generally point me at documentation?

Reply:
If Karma directly uses `BABELIFY_PLUGINS` and all tests pass with this PR, I think that's good enough.

I was trying to link to a way to print the list of files being transformed. I guess that link was the wrong one. There are two more Karma tricks that might help:

1) Change `ERROR` to `DEBUG` to see what Karma is doing
https://github.com/ampproject/amphtml/blob/286109abe0d2241c92407f3fac7b6d48e9ed2a2a/build-system/tasks/karma.conf.js#L139

2) Change `process.stdout.write('.')` here to `console.log(<file being transformed>)`
https://github.com/ampproject/amphtml/blob/286109abe0d2241c92407f3fac7b6d48e9ed2a2a/build-system/tasks/runtime-test/runtime-test-base.js#L198-L204

@jridgewell does this SGTY?